Fix has_command, CentOS detection, source installation and add OS update support

avsm opened the pull request:
```
$ command -v fooooo
$ echo $?
1
$ /bin/sh -c command -v foooo
$ echo $?
0
```

We now just call `command` directly, which fixes #6

avsm committed:
Fix has_command by not going through a subshell

```
$ command -v fooooo
$ echo $?
1
$ /bin/sh -c command -v foooo
$ echo $?
0
```

We now just call `command` directly, which fixes #6

avsm committed:
bump the working version of depext to 0.4

avsm committed:
Fix CentOS depext

The distribution name is "CentOS", not "Centos". Just force lowercase
comparison to simplify this, as this may have varied between CentOS
6 and 7.

Also improve the error message to distinguish between an unknown
distribution and other parsing error.

avsm committed:
Do not double-escape in run_command, which fixes source depexts

dbuenzli commented on depext.ml:
```diff
@@ -41,11 +41,11 @@ let string_split char str =
 
 let has_command c =
-  let cmd = Printf.sprintf "/bin/sh -c command -v %s" c in
+  let cmd = Printf.sprintf "command -v %s" c in
```
I'm not sure this will work well. Based on experience with cmdliner [that's](https://github.com/dbuenzli/cmdliner/blob/master/src/cmdliner.ml#L349-L355) a correct and multiplatform way of checking whether a command exists or not.

avsm replied:
`command` is POSIX, but I can believe that it's not portable despite that.  Would it be possible for `cmdliner` to expose the `find_cmd` logic so that we dont have to cut and paste it here?

dbuenzli replied:
> Would it be possible for cmdliner to expose the find_cmd logic so that we dont have to cut and paste it here?

I don't think `cmdliner` is the place to expose such things. I'd say either just c&p or use Bos which [exposes](http://erratique.ch/software/bos/doc/Bos.OS.Cmd.html#VALexists) this logic in its API (but that's another dependency).

avsm committed:
Add `-u` to trigger an OS update before external dependency installation.

This is useful on Debian or Ubuntu where package installation may fail if the
sets are not up-to-date with respect to latest security updates.

avsm committed:
Add a CHANGES file

avsm commented:
This pull request has become a bit feature packed :)  I'm using it in the bulk builds with all this now though...

avsm committed:
fix typo in opam file

AltGr commented:
Awesome! :)
